contracts/PreToken.sol

```diff
@@ -61,27 +62,55 @@ contract PreToken is Restricted {
         emit Transfer(0x0, _to, amount);
     }
 
+    /* Restricted function to allow pretoken signers to fix incorrect issuance */
+    function burnFrom(address from, uint amount)
+    public restrict("PreTokenSigner") returns (bool) {
+        require(amount > 0, "burn amount must be > 0"); // this effectively restricts burning from agreement holders only
+        require(agreements[from].balance >= amount, "must not burn more than balance"); // .sub would revert anyways but emit reason
```

Comment:
not sure I see the use-case for a partial revocation. I would make this either binary (burn the entire agreement/amount), or provide the ability to change all the fields arbitrarily.

Reply:
Right, but you could have made a mistake in either direction or issued the cap incorrectly...

Reply:
if we made a mistake in the cap then we burn the whole balance and create a new agreement to a new account
if we didn't issue enough then we just issue more

If we would only have the option to burn the whole balance then the issue / transfer history would be very convoluted for an account if we would reissue to the same account.  Or If we reissue to a new account always then the history would be lost even if we just made one issueTo mistake.